sites/all/modules/custom/training/training.pages.inc

```diff
@@ -0,0 +1,241 @@
+    'Argument 2 = <em>' . $two . '<em>';
+}
+
+/**
+ * Page callback for training_menu(),
+ * called from $menu['menu/placeholder/%/display']
+ *
+ * @param $smth
+ *  Any value user enters instead of % in menu/placeholder/%/display
+ * @return mixed
+ */
+function training_placeholder_page($smth){
+  return $smth;
+}
+
+/**
+ * Page callback for training_menu(),
+ * called from $menu['menu/default_arg/%']
+ *
+ * @param $arg
+ *  Any value user enters instead of % in menu/default_arg/%
+ * @return string
+ */
+function training_default_arg_page($arg){
+  switch ($arg) {
+    case 1:
+      return 'one';
+      break;
+    case 2:
+      return 'two';
+      break;
+    case 3:
+      return 'three';
+      break;
+    case 99:
+      return 'Almost there';
+      break;
+    default:
+      return '404 page is not found';
+  }
+}
+
+/**
+ * Title callback for training_menu(),
+ * called from $menu['menu/title']
+ * displays for logged user - user name, for not logged - anonymouse
+ *
+ * @return string
+ */
+
+function training_title_title(){
+  global $user;
+  if (user_is_logged_in()){
+    $username = $user->name;
+  }
+  else {
+    $username = 'anonymouse';
+  }
+
+  return 'Dynamic title: username = ' . $username;
+}
+
+/**
+ * Page callback for training_menu(),
+ * called from $menu['theme'], $menu['theme/link']
+ *
+ * @param string $atr
+ *  parameter to  specify
+ * @return string
+ * @throws \Exception
+ */
+function training_theme_page($atr = 'parent'){
+  $links = array('list','table','image','access');
+  if ($atr == 'parent'){
+    foreach ($links as &$link){
+      $link = 'theme/' . $link;
+    }
+  }
+  return theme('training_theme',array(
+    'title' => 'Links to themes',
+    'items' => array('List','Table','image','Access'),
+    'links' => $links,
+  ));
+}
+
+/**
+ * Page callback for training_menu(),
+ * called from $menu['theme/table']
+ *
+ * @return string
+ * @throws \Exception
+ */
+function training_theme_table_page(){
+  $title = '<h1' .
+    drupal_attributes(array('id' => 'table-test')) .
+    '>Theme table sample<br>';
+  $header = array ('header 1','header 2', 'header 3');
+  $rows=array(
+    array(
+      'data'=> array(
+        '11',
+        array('data'=>'12','colspan'=>2),
+      ),
+      'class' => array('rowClass')),
+    array(
+      'data' => array('21','22','23'),
+      'class' => array('rowClass')
+    ),
+    array(
+      'data' => array('21','22','23'),
+      'class' => array('rowClass')
+    ),
+  );
+  return $title . theme('table',[
+    'header' => $header,
+    'rows' => $rows,
+    'attributes' => array('id' =>'table-test'),
+  ]);
+}
+
+/**
+ * Page callback for training_menu(),
+ * called from $menu['theme/image']
+ *
+ * @return string
+ * @throws \Exception
+ */
+function training_theme_image_page(){
+  $path =  drupal_get_path('module', 'training') . '/image/sample.png';
+  $ar = image_styles();
+  $keys = array_keys($ar);
+
+  foreach ($ar as $i => $style){
```

Comment:
Here I've optimized the code, but it still does not work - image with implemented style (by theme_image_style) is not visible.